metagraph/src/graph/alignment/dbg_aligner.hpp

```diff
@@ -24,377 +20,153 @@ class IDBGAligner {
     typedef QueryAlignment<node_index> DBGQueryAlignment;
     typedef typename DBGAlignment::score_t score_t;
 
-    virtual ~IDBGAligner() {}
+    typedef std::function<void(std::string_view /* header */,
+                               std::string_view /* seq */,
+                               bool /* orientation of seq */)> QueryCallback;
+    typedef std::function<void(const QueryCallback&)> QueryGenerator;
+    typedef std::function<void(std::string_view /* header */,
+                               DBGQueryAlignment&& /* alignments */)> AlignmentCallback;
 
-    virtual DBGQueryAlignment align(const std::string_view query) const = 0;
+    virtual ~IDBGAligner() {}
 
-    virtual const DeBruijnGraph& get_graph() const = 0;
-    virtual const DBGAlignerConfig& get_config() const = 0;
+    virtual void align_batch(const QueryGenerator &generate_query,
+                             const AlignmentCallback &callback) const = 0;
+    DBGQueryAlignment align(std::string_view query,
+                            bool is_reverse_complement = false) const;
 };
 
-template <class Seeder = ExactSeeder<>,
-          class Extender = DefaultColumnExtender<>>
-class SeedAndExtendAligner : public IDBGAligner {
+template <class AlignmentCompare = std::less<IDBGAligner::DBGAlignment>>
```

Comment:
this template is not used anywhere

Reply:
it's used in the implementation of AlignAggregate